Use feature flag or lab feature for showing bnmo features

sweir27 opened the pull request:
This PR follows up https://github.com/artsy/force/pull/2958 by adding a check for the `ENABLE_NEW_BUY_NOW_FLOW` feature flag everywhere we are currently checking for the lab feature.

Verified locally (desktop + mobile web) with artworks in the following states:
- Gallery artwork enabled for ecommerce
- Auction Partner artwork enabled for ecommerce
- Artwork in an auction enabled for ecommerce + bidding

Once this is merged, we should set the flag to TRUE on staging. Note that if you are logged out, you won't be able to create an order until https://artsyproduct.atlassian.net/browse/PURCHASE-487 is done.

mzikherman commented on src/mobile/apps/artwork/components/meta_data/view.coffee:
```diff
@@ -21,7 +21,7 @@ module.exports = class MetaDataView extends Backbone.View
       @model.get('partner').type == 'Auction House'
 
-    if loggedInUser?.hasLabFeature('New Buy Now Flow')
+    if sd.ENABLE_NEW_BUY_NOW_FLOW || loggedInUser?.hasLabFeature('New Buy Now Flow')
```
do you need a `sd = require('sharify').data` at the top here?

mzikherman commented on src/mobile/apps/artwork/templates/index.jade:
```diff
@@ -7,7 +7,7 @@ block content
   include ../components/image/templates/index.jade
   include ../components/meta_data/templates/index.jade
-  if !user || !user.hasLabFeature('New Buy Now Flow')
+  if !sd.ENABLE_NEW_BUY_NOW_FLOW && !(user || user.hasLabFeature('New Buy Now Flow'))
```
I think it should be `!(user && user.hasLabFeature('New Buy Now Flow'))` (`&&` instead of `||` in the lab feature check)

mzikherman commented:
Some minor comments, looks like real ❌ tho. Merge when ready!

sweir27 commented:
Updated this to add explicit parens around the lengthy logic, fixed up an erroneous `||`, and fixed the tests that were failing due to `sd` needing to be passed in.
